src/parser.jl

```diff
@@ -3,11 +3,15 @@
 isevaldef(x) = Base.Meta.isexpr(x, :(=)) && Base.Meta.isexpr(x.args[1], :call) &&
                x.args[1].args[1] == :eval
 
-# This detects two types of function declarations: those that use the
-# `function` keyword (first case) and those that are defined as
-# `f() = expr` (the second case).
+# This detects different types of function declarations:
+# - those that use the  `function` keyword;
+# - those that are defined as `f = x -> expr`
+# - those that are defined as `f() = expr`
+# - those that are defined as `f() where {T} = expr`
 isfuncexpr(ex::Expr) =
-    ex.head == :function || (ex.head == :(=) && typeof(ex.args[1]) == Expr && ex.args[1].head == :call)
+    ex.head == :function || ex.head == :-> ||
+        (ex.head == :(=) && typeof(ex.args[1]) == Expr &&
+            (ex.args[1].head == :call || (ex.args[1].head == :where && ex.args[1].args[1].head == :call)))
```

Comment:
Could also use `Base.is_short_function_def` for part of this (to handle a few more cases, including `f() where A where B = c` and `()::C = c`)

Reply:
Cool, didn't know about `Base.is_short_function_def`. This indeed handles `f() where A where B = c`. However, it does not seem to recognize `()::C = c` as a function definition either.